File: go/grouping.go
package main

import "fmt"

type Product struct {
    Name     string
    Category string
    Price    int64
}

func main() {
    products := [...]Product{
        {Name: "Laptop", Category: "Electronics", Price: 99999},
        {Name: "Smartphone", Category: "Electronics", Price: 49999},
        {Name: "T-Shirt", Category: "Clothing", Price: 1999},
        {Name: "Jeans", Category: "Clothing", Price: 3999},
        {Name: "Coffee Maker", Category: "Home Appliances", Price: 8999},
    }

    groupedProducts := map[string][]Product{}
    for _, p := range products {
        groupedProducts[p.Category] = append(groupedProducts[p.Category], p)
    }

    for c, g := range groupedProducts {
        fmt.Println("Category:", c)
        for _, p := range g {
            fmt.Printf(" - %s: $%d.%02d\n", p.Name, p.Price/100, p.Price%100)
        }
    }
}

File: kotlin/src/Grouping.kt

data class Product(
    val name: String,
    val category: String,
    val price: Double,
)

fun main() {
    val products = listOf(
        Product("Laptop", "Electronics", 999.99),
        Product("Smartphone", "Electronics", 499.99),
        Product("T-Shirt", "Clothing", 19.99),
        Product("Jeans", "Clothing", 39.99),
        Product("Coffee Maker", "Home Appliances", 89.99),
    )

    val groupedProducts = products.groupBy { it.category }

    groupedProducts.forEach { group ->
        println("Category: ${group.key}")
        group.value.forEach { product ->
            println(" - ${product.name}: $${product.price}")
        }
    }
}